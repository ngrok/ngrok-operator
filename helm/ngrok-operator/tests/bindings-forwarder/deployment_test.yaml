suite: test bindings forwarder deployment
templates:
- bindings-forwarder/deployment.yaml
- bindings-forwarder/rbac.yaml
set:
  bindings.enabled: true
tests:
- it: Should match snapshot
  template: bindings-forwarder/deployment.yaml
  asserts:
  - matchSnapshot: {}
- it: should create a deployment if bindings.enabled is true
  template: bindings-forwarder/deployment.yaml
  asserts:
  - isKind:
      of: Deployment
  - isAPIVersion:
      of: apps/v1
  - equal:
      path: metadata.name
      value: RELEASE-NAME-ngrok-operator-bindings-forwarder
