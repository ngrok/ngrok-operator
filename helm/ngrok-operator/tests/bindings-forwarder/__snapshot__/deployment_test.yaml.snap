Should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        checksum/rbac: aed9a6319200708b35eb337a6d4e136684fbd52eae02297565c8e37db9d99ba4
      labels:
        app.kubernetes.io/component: bindings-forwarder
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: ngrok-operator
        app.kubernetes.io/part-of: ngrok-operator
        app.kubernetes.io/version: 0.14.1
        helm.sh/chart: ngrok-operator-0.17.0-rc.2
      name: RELEASE-NAME-ngrok-operator-bindings-forwarder
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: bindings-forwarder
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: ngrok-operator
      template:
        metadata:
          annotations:
            checksum/rbac: aed9a6319200708b35eb337a6d4e136684fbd52eae02297565c8e37db9d99ba4
            prometheus.io/path: /metrics
            prometheus.io/port: "8080"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/component: bindings-forwarder
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: ngrok-operator
        spec:
          affinity:
            nodeAffinity: null
            podAffinity: null
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution:
                - podAffinityTerm:
                    labelSelector:
                      matchLabels:
                        app.kubernetes.io/component: bindings-forwarder
                        app.kubernetes.io/instance: RELEASE-NAME
                        app.kubernetes.io/name: ngrok-operator
                    topologyKey: kubernetes.io/hostname
                  weight: 1
          containers:
            - args:
                - --release-name=RELEASE-NAME
                - --description="The official ngrok Kubernetes Operator."
                - --zap-log-level=info
                - --zap-stacktrace-level=error
                - --zap-encoder=json
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8080
                - --manager-name=RELEASE-NAME-ngrok-operator-bindings-forwarder
              command:
                - /bindings-forwarder-manager
              env:
                - name: NGROK_AUTHTOKEN
                  valueFrom:
                    secretKeyRef:
                      key: AUTHTOKEN
                      name: RELEASE-NAME-ngrok-operator-credentials
                - name: POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: HELM_RELEASE_NAME
                  value: RELEASE-NAME
                - name: SSL_CERT_DIR
                  value: /etc/ssl/certs/ngrok
              image: docker.io/ngrok/ngrok-operator:0.14.1
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: forwarder
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
              volumeMounts:
                - mountPath: /etc/ssl/certs/ngrok
                  name: ngrok-ca
                  readOnly: true
          serviceAccountName: RELEASE-NAME-ngrok-operator-bindings-forwarder
          volumes:
            - configMap:
                name: ngrok-intermediate-ca
              name: ngrok-ca
